### ISSUE TITLE
Vulkan’s coordinate system

### ISSUE THREAD

**Issue by amerkoleci** [mellinoe/veldrid#8]
Hi dude,
I saw you're adding vulkan support and probably found yourself in the y inverted position where you should whether flip the gl_Position.y or handle something else.

I found this article where it explain some stuff about vulkan coordinate system:

http://anki3d.org/vulkan-coordinate-system/

**Comment by mellinoe** [comment]
Hah! It's funny that you posted this -- I was just researching this earlier today. Right now I am just inverting the y-coordinate in my shader like you mentioned. This `KHR_VK_maintainance1` extension seems like it will be a better solution, and for some reason it didn't show up in any of then links I found. But I'm not sure how reasonable it is to assume it's always available. I may just assume that it is while I am bringing the Vulkan backend online, just to make things easier...

This problem is similar to the difference in clip space Z-range between OpenGL and D3D11. The Matrix4x4 type in the BCL produces z values in the 0->1 range, directly usable for D3D transformations, but needing a fixup for OpenGL. I fixed the sample code here to account for that in [this commit](https://github.com/mellinoe/veldrid/commit/843017cc343b11793c0ff582919391d98a34eb68). I actually didn't understand this problem for a long time, and ended up finishing CrazyCore with incorrect depth handling on OpenGL 😆.

If I understand correctly, I think I should be able to directly use Matrix4x4 in the Vulkan backend as long as I rely on this KHR_VK_maintainance1 extension, and correctly set an inverted viewport.
